cg/models/orders/samples.py

```diff
@@ -105,11 +121,15 @@ def required_for_new_samples(cls, value, values, **kwargs):
     ]
     synopsis: Optional[str]
 
-    @validator("tumour_purity", "formalin_fixation_time", "post_formalin_fixation_time", pre=True)
+    @validator(
+        "tumour_purity",
+        "formalin_fixation_time",
+        "post_formalin_fixation_time",
+        "quantity",
+        pre=True,
+    )
     def str_to_int(cls, v: str) -> Optional[int]:
-        if not v:
-            return None
-        return int(v)
+        return OptionalIntValidator.str_to_int(v=v)
```

Comment:
Maybe add str to float here to handle it for all use-cases? Even if you validate different fields in different orderforms, it still validates None to None if the value is not filled

Reply:
It was my intuition to put these in the base class but I haven't figured out how to decorate them in a prettier way than this the derived classes.

Reply:
I think I misunderstood what you meant. Now I think you meant that we should use the str_to_float(v=v) to convert both ints and floats to float!?